api/agent/agent.go

```diff
@@ -541,6 +546,11 @@ func (a *agent) runHot(ctxArg context.Context, slots chan<- slot, call *call, to
 		stdin:  stdinRead,
 		stdout: stdoutWrite,
 		stderr: &ghostWriter{inner: stderr},
+
+		ds:      a.ds,
+		appName: call.AppName,
+		path:    call.Path,
+		callID:  call.ID,
```

Comment:
none of these should be on the container, really. the context has a logger with these values, but adding a note to be thorough, see comment below about contract for writing stats.